DylibBinder/InnerX.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SwiftReflector.Inventory;
+
+namespace DylibBinder {
+	public class InnerX {
+		public InnerX ()
+		{
+			dict = new Dictionary<string, List<ClassContents>> ();
+		}
+
+		Dictionary<string, List<ClassContents>> dict { get; }
+
+		public Dictionary<string, List<ClassContents>> AddClassContentsList (params List<ClassContents>[] contents)
+		{
+			foreach (var classContentsList in contents) {
+				foreach (var classContent in classContentsList) {
+					AddItem (classContent);
+				}
+			}
+			return dict;
+		}
+
+		void AddItem (ClassContents c)
+		{
+			var nestingNames = c.Name.NestingNames.ToList ();
+			var nestingNameString = ConvertNestingNamesToString (nestingNames);
+
+			if (nestingNameString == null)
+				return;
+
+			if (nestingNames.Count == 1) {
+				AddKeyIfNotPresent (nestingNameString);
+				return;
+			}
+
+			var parentNestingNames = nestingNames;
+			parentNestingNames.RemoveAt (parentNestingNames.Count - 1);
```

Comment:
Also good call! I changed that null check to a ==0 check because although there should always be at least one item (as far as I understand), nestingNames can't be null so creating a list from it should not be null.